packages/mgt-components/src/components/mgt-people/mgt-people.ts

```diff
@@ -449,8 +449,6 @@ export class MgtPeople extends MgtTemplatedComponent {
           this.people = await getPeople(graph);
         }
 
-        this.people = this.people.filter(person => person !== null && person !== '');
```

Comment:
@gavinbarron, I have checked this out _it shouldn't be removed_(?).

I have figured out that Waldek made a fix for a scenario however, there's a problem that still exists with fallback details, and I have an idea on how to fix it without impacting Waldek's _expectations_.

Currently, `mgt-people`, has the `fallback-details` and the `people-queries` attributes. When they are used together, we get into this `null` people _problem_.

Line 434 will try to fetch the people given queries correctly:
https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/3034552fbfc7b58457b2d82b8aaeefee4a309876/packages/mgt-components/src/components/mgt-people/mgt-people.ts#L432-L434

The implementation of `getUsersForPeopleQueries` will return no values in the `BatchResponse` if the query is a value that does not exist in the graph. The initial implementation sets a `null` as part of the returned `people` array **(problem 1)**.

https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/73dc57e4584d6dca4297a25eaa73751a5c8d77dd/packages/mgt-components/src/graph/graph.user.ts#L311-L313

Lastly, the fallback details will be used to replace the `null` values in the `people` array **(problem 2)**:

https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/3034552fbfc7b58457b2d82b8aaeefee4a309876/packages/mgt-components/src/components/mgt-people/mgt-people.ts#L435-L444

### Issues with all these...
- `null` values will exist when `people-queries` are used and `fallback-details` are not set. Mainly caused by problem 1 which Waldek fixed by filtering out `nulls`.
- The assigning of `fallback-details` to `null` people values are done linearly on an index basis. This approach brings a scenario where `people-queries` have to be aligned with their corresponding `fallback-details`. Which means a little mismatch like the fallback details being less than the people queries cause an error. Try removing one fallback detail block here https://mgt.dev/?path=/story/components-mgt-people-properties--fallback-details-people-query.

#### Problem 1 possible solution
- Return results which have actual values and ignore responses without values. This warrants removal of the line above but now you can't set fallback details.

#### Problem 2 possible solution
- Change how we setup the fallback details. We match the people query that had no values to the values supplied for the fallback details and set the fallback details to the values that match. This should be done when a call to graph returns no values. The `getUsersForPeopleQueries` function will be updated to pass in the fallback details but I don't think that it will impact any performance or usages. SOmething like:
```javascript
...
if (batch.hasRequests) {
    try {
      const responses = await batch.executeAll();

      for (const personQuery of peopleQueries) {
        const response = responses.get(personQuery);
        if (response && response.content && response.content.value && response.content.value.length > 0) {
          people.push(response.content.value[0]);
          if (getIsUsersCacheEnabled()) {
            cache.putValue(personQuery, { maxResults: 1, results: [JSON.stringify(response.content.value[0])] });
          }
        }else{
          fallbackDetails.forEach(detail=>{
            const hasFallBack = Object.values(detail).includes(personQuery);
            if(hasFallBack){
              people.push(detail);
            }
          })
        }

      return people;
    } catch (_) {}
...
```

What do you think about this?

Reply:
Thank you for the great write up here on the extended issue, I really appreciate the effort you put into investigating and reporting this.

I like the second solution as it seems a bit more robust.

The else clause should probably use a find to get the matching fallback rather than a foreach on each pass. 

```typescript
        }else{
          const fallback = fallbackDetails.find(detail=> Object.values(detail).includes(personQuery));
          if(fallback){
            people.push(fallback);
          }
        }
```